packages/loopring_v3/contracts/aux/access/LoopringIOExchangeOwner.sol

```diff
@@ -114,10 +116,11 @@ contract LoopringIOExchangeOwner is SelectorBasedAccessManager, ERC1271, Drainab
             "INVALID_DATA"
         );
 
+        // Decode the blocks
+        ExchangeData.Block[] memory blocks = _decodeBlocks(decompressed);
```

Comment:
Brecht, maybe we should check the 'approved' field in the aux data must always be FALSE inside _decodeBlocks. (or we simply assume they are all force without loading them). Then in _beforeBlockSubmission, we allow the callbacks to change some tx's 'approved' field to true.

Inside the Loopring protocol, if approved == true, we skip, otherwise, we through.

Reply:
I think that's basically what is done. The intermediate array `preApprovedTxs` is used to make this easier with values that are new'd are initialized to zero and then values there are set to true. And then all those values are checked against the data passed in without modifying any block data in the smart contracts. We could start modifying the data in the block, but seems less clean to do it like that. Unless this would make the relayer much simpler, not having to set the correct `approved` value, but I would think not really.